### ISSUE TITLE
False-positive `type mapping mismatch` when string literals and ENUM are used together

### ISSUE THREAD

**Issue by fluffycondor** [phpstan/phpstan-doctrine#225]
phpstan-doctrine 1.0.0

````php
/**
 * @var 'A'|'B'
 * @ORM\Column(type="string", length=255, columnDefinition="ENUM('A', 'B')")
 */
private string $type;
````
It leads to a false-positive error `Property Foo::$type type mapping mismatch: database can contain string but property expects 'A'|'B'. `

**Comment by ondrejmirtes** [comment]
This is expected. Someone can insert `'C'` into your database. You need a special Doctrine type.
